shellai/handlers.py

```diff
@@ -65,54 +25,48 @@ def handle_script_session(shellai_tmp_file) -> None:
         os.remove(shellai_tmp_file)
 
 
-def _handle_caret(query: str, config: dict) -> str:
+def _handle_caret(query: str, config: Config) -> str:
     """
     Replaces caret (^) with command output specified in config file.
     """
     if "^" not in query:
         return query
 
-    output_capture_settings = config.get("output_capture_settings", {})
-    captured_output_file = output_capture_settings.get(
-        "captured_output_file", "/tmp/shellai_output.txt"
-    )
+    captured_output_file = config.output.file
 
     if not os.path.exists(captured_output_file):
         logging.error(
             f"Output file {captured_output_file} does not exist, change location of file in config to use '^'."
         )
         exit(1)
 
-    prompt_separator = output_capture_settings.get("prompt_separator", "$")
+    prompt_separator = config.output.prompt_separator
     with open(captured_output_file, "r") as f:
         # NOTE: takes only last command + output from file
         output = f.read().split(prompt_separator)[-1].strip()
+
     query = query.replace("^", "")
     query = f"Context data: {output}\nQuestion: " + query
     return query
 
 
-def handle_query(query: str, config: dict) -> None:
+def handle_query(query: str, config: Config) -> None:
     query = _handle_caret(query, config)
     # NOTE: Add more query handling here
 
     logging.info(f"Query: {query}")
 
-    backend_service = config.get("backend_service", {})
-    query_endpoint = backend_service.get(
-        "query_endpoint", "http://0.0.0.0:8080/v1/query/"
-    )
+    query_endpoint = config.backend.endpoint
 
     try:
-        history_conf = config.get("history", {})
-        history = _handle_history_read(history_conf)
+        history = handle_history_read(config)
         payload = get_payload(query)
         logging.info("Waiting for response from AI...")
         response = requests.post(
             query_endpoint,
             headers={"Content-Type": "application/json"},
             data=json.dumps(payload),
-            timeout=30,  # waiting for more than 30 seconds does not make sense
+            timeout=320,  # waiting for more than 30 seconds does not make sense
```

Comment:
Since we're increasing this from 30 to 320, should we remove the comment?

Reply:
Hmm, that's a good question. I'm hitting the timeout with OLS with the 30 seconds value. I might need to have some round of tests to figure out a good value here.

Reply:
The timeout is fine! I was just asking about the comment. It looks like this one needs a rebase, too.

Reply:
Thinking more about this, I will revert for now and open a ticket for us to figure out the ideal timeout value here.